src/components/cart/CartItem.jsx

```diff
@@ -0,0 +1,122 @@
+import { faCartShopping } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React from "react";
+import { useAuth, useData } from "../../contexts";
+import { updateQuantityInCart, removeFromCart } from "../../services";
+import { getOriginalPrice } from "../productListing/Products";
+import "./cart.css";
+
+function CartItem() {
+  const { state, dispatch } = useData();
+  const { authToken } = useAuth();
+
+  const updateQuantity = async (item, type) => {
+    let response;
+    if (type === "increment")
+      response = await updateQuantityInCart(item._id, authToken, type);
+    else {
+      if (item.qty === 1) response = await removeFromCart(item._id, authToken);
+      else response = await updateQuantityInCart(item._id, authToken, type);
+    }
+    dispatch({ type: "CART_OPERATION", payload: { cart: response.cart } });
+  };
+
+  const removeItem = async (item) => {
+    const response = await removeFromCart(item._id, authToken);
+    dispatch({ type: "CART_OPERATION", payload: { cart: response.cart } });
```

Comment:
CART_OPERATION can be inside some constant variable so you don't have to change throughout your app

Reply:
Sure, will implement this for all constants in refactor branch